mycore-base/src/main/java/org/mycore/common/MCRUserInformationResolver.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * This file is part of ***  M y C o R e  ***
+ * See http://www.mycore.de/ for details.
+ *
+ * MyCoRe is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * MyCoRe is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MyCoRe.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.mycore.common;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.status.StatusLogger;
+import org.mycore.common.config.MCRConfiguration2;
+import org.mycore.common.config.MCRConfigurationException;
+import org.mycore.common.log.MCRListMessage;
+
+/**
+ * A {@link MCRUserInformationResolver} can be used to obtain {@link MCRUserInformation}, without knowledge
+ * of the underlying mechanism that creates or looks up that user information, by providing a string specification.
+ * The specification can be created from a schema and a user ID with
+ * {@link MCRUserInformationResolver#getSpecification(String, String)}.
+ * <p>
+ * Multiple instances of {@link MCRUserInformationProvider} can be configured as a comma separated list of fully
+ * qualified class names with the configuration property {@link MCRUserInformationResolver#PROVIDERS_KEY}.
+ * Each user information provider has a fixed schema and implements a strategy to create or look up user information
+ * for a given user ID.
+ * <p>
+ * A possible use case is to have the ability of storing the user information required for e.g. 
+ * {@link org.mycore.util.concurrent.MCRFixedUserCallable} as a string in a configuration property.
+ */
+public final class MCRUserInformationResolver {
+
+    public static final String PROVIDERS_KEY = "MCR.UserInformation.Resolver.Providers";
+
+    private static final Logger LOGGER = StatusLogger.getLogger();
+
+    private static final MCRUserInformationResolver INSTANCE = new MCRUserInformationResolver(getProviders());
+
+    private final Map<String, MCRUserInformationProvider> providers;
+
+    public MCRUserInformationResolver(Map<String, MCRUserInformationProvider> providers) {
+        this.providers = Objects.requireNonNull(providers);
+    }
+
+    private static Map<String, MCRUserInformationProvider> getProviders() {
+
+        List<MCRUserInformationProvider> providers = MCRConfiguration2
```

Comment:
Maybe it would be better if we use org.mycore.common.config.MCRConfiguration2#getInstances .

1. It would be the same principle as uri resolvers
2. It would avoid the issue that Providers can have the same schema
3. It would make it easier to exchange a specific provider
4. It would make the getProviders Method easier. 
5. The getSchema method could be removed

Reply:
I'll look into that.